open-build-service-mock/src/api/build.rs

```diff
@@ -309,6 +310,132 @@ impl Respond for BuildResultsResponder {
     }
 }
 
+pub(crate) struct BuildJobHistoryResponder {
+    mock: ObsMock,
+}
+
+impl BuildJobHistoryResponder {
+    pub fn new(mock: ObsMock) -> Self {
+        Self { mock }
+    }
+}
+
+impl Respond for BuildJobHistoryResponder {
+    fn respond(&self, request: &Request) -> ResponseTemplate {
+        try_api!(check_auth(self.mock.auth(), request));
+
+        let mut package_names = vec![];
+        let mut code_names = vec![];
+        let mut limit = None;
+
+        for (key, value) in request.url.query_pairs() {
+            match key.as_ref() {
+                "package" => package_names.push(value.into_owned()),
+                "code" => code_names.push(value.into_owned()),
+                "limit" if limit.is_some() => {
+                    return ApiError::new(
+                        StatusCode::Ok,
+                        "400".to_owned(),
+                        "parameter 'limit' set multiple times".to_owned(),
+                    )
+                    .into_response()
+                }
+                "limit" => limit = Some(try_api!(parse_number_param(value))),
+                _ => return unknown_parameter(&key).into_response(),
+            }
+        }
+
+        let mut components = request.url.path_segments().unwrap();
+        let arch = components.nth_back(1).unwrap();
+        let repo_name = components.nth_back(0).unwrap();
+        let project_name = components.nth_back(0).unwrap();
+
+        let projects = self.mock.projects().read().unwrap();
```

Comment:
It looks like this lock is held for far longer than necessary?

Reply:
Not really, since stuff like `arch` is all references, so the original lock needs to be held.